generator/generic_generator_utilities.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+module Inferno
+  module Generator
+    module GenericGeneratorUtilties
+      EXPECTATION_URL = 'http://hl7.org/fhir/StructureDefinition/capabilitystatement-expectation'
+
+      def structure_to_string(struct)
+        if struct.is_a? Hash
+          %({
+            #{struct.map { |k, v| "#{k}: #{structure_to_string(v)}" }.join(",\n")}
+          })
+        elsif struct.is_a? Array
+          if struct.empty?
+            '[]'
+          else
+            %([
+              #{struct.map { |el| structure_to_string(el) }.join(",\n")}
+            ])
+          end
+        elsif struct.is_a? String
+          "'#{struct}'"
+        elsif [true, false].include? struct
+          struct.to_s
+        else
+          "''"
+        end
+      end
+
+      def create_search_validation(sequence_metadata)
+        search_validators = ''
+        sequence_metadata.search_parameter_metadata&.each do |parameter_metadata|
+          type = sequence_metadata.element_type_by_path(parameter_metadata.expression) || parameter_metadata.type
+          path = parameter_metadata.expression
+            .gsub(/(?<!\w)class(?!\w)/, 'local_class')
+            .split('.')
+            .drop(1)
+            .join('.')
+          path += get_value_path_by_type(type) unless ['Period', 'date', 'HumanName', 'Address', 'CodeableConcept', 'Coding', 'Identifier'].include? type
+          parameter_code = parameter_metadata.code
+          resource_type = sequence_metadata.resource_type
+          search_validators += %(
+              when '#{parameter_metadata.code}'
+              values_found = resolve_path(resource, '#{path}')
+              #{search_param_match_found_code(type, parameter_metadata.code)}
+              assert match_found, "#{parameter_code} in #{resource_type}/\#{resource.id} (\#{values_found}) does not match #{parameter_code} requested (\#{value})"
+            )
+        end
+
+        validate_functions =
+          if search_validators.empty?
+            ''
+          else
+            %(
+              def validate_resource_item(resource, property, value)
+                case property
+                  #{search_validators}
+                end
+              end
+            )
+          end
+        validate_functions
+      end
+
+      def search_param_match_found_code(type, element)
```

Comment:
Rather than duplicating this code in each sequence, can you stick this method's logic (without all the stringifying) into `SequenceUtilities`.

Reply:
I think this is reasonable. But I do remember Rob saying something about writing out functions explicitly like this in each sequence to make it easier to understand. But anyway, can we leave this to a future PR?

Reply:
My preference in general is for it to be fairly obvious what a given test does without having to continually hop to other shared code locations.  Yeah, it results in duplication of generated code, but that duplicated code never needs to managed indivdually (its managed as a group in the generator).  Just a preference.  But if we end up with a lot of tests that basically are just a call to a shared function that needs to be followed to understand what is going on at all i think we are doing it wrong.